sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -216,19 +221,32 @@ def check_two_point(r, dualtree):
 
 
 def test_ball_tree_pickle():
-    import pickle
     np.random.seed(0)
     X = np.random.random((10, 3))
+
     bt1 = BallTree(X, leaf_size=1)
+    # Test if BallTree with callable metric is picklable
+    bt1_pyfunc = BallTree(X, metric=dist_func, leaf_size=1, p=2)
+
     ind1, dist1 = bt1.query(X)
+    ind1_pyfunc, dist1_pyfunc = bt1_pyfunc.query(X)
 
     def check_pickle_protocol(protocol):
         s = pickle.dumps(bt1, protocol=protocol)
         bt2 = pickle.loads(s)
+
+        s_pyfunc = pickle.dumps(bt1_pyfunc, protocol=protocol)
+        bt2_pyfunc = pickle.loads(s_pyfunc)
```

Comment:
Yes! but this particular line will pass, querying alone will fail. (new line #+242)